analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/SecureRandomSeedsShouldNotBePredictable.cs

```diff
@@ -145,6 +171,5 @@ private ProgramState ProcessNextMethods(ProgramState state, IInvocationOperation
         invocation.TargetMethod.ContainingType.Is(KnownType.Org_BouncyCastle_Security_SecureRandom);
 
     private static bool IsIRandomGenerator(IInvocationOperationWrapper invocation) =>
-        invocation.Instance is { } instance
-        && instance.Type.Is(KnownType.Org_BouncyCastle_Crypto_Prng_IRandomGenerator);
+        invocation.TargetMethod.ContainingType.DerivesOrImplements(KnownType.Org_BouncyCastle_Crypto_Prng_IRandomGenerator);
```

Comment:
This method is called during _seeding_ and during _generating_ a random value.
Technically, all three of these are valid:

```csharp
IRandom/Digest/Vmprc thingy = new Digest/Vmprc(...);
thingy.NextBytes(bytes); // thingy can be any of the three here
```

So to make sure we can handle all three of the variants, I check the more broad question "does thingy implement IRandom".

During object creation, I care specifically about the concrete `Digest` and `Vmprc`, so we cannot get here with a custom, different implementation.